File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    f = [0,0]
    for i in keyinput:
        if i == "right" and f[0]<(board[0]-1)//2:
            f[0] += 1
        elif i == "left" and f[0]>-((board[0]-1)//2):
            f[0] -= 1
        elif i == "up" and f[1]<(board[1]-1)//2:
            f[1] += 1
        elif i == "down" and f[1]>-((board[1]-1)//2):
            f[1] -= 1
    return f

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
   public int[] solution(String[] keyinput, int[] board) {
    int[] answer = new int[2];
    for (String i : keyinput){
        if (i.equals("left")) {
                if (answer[0] > -1 * board[0]/2){
                    answer[0] -= 1;
                }
            }

            
        if (i.equals("right")) {
                if (answer[0] < board[0]/2){
                    answer[0] += 1;
                }
        }
        
        if (i.equals("up")) {
                if (answer[1] < board[1]/2){
                    answer[1] += 1;
                }
        }
            
        if (i.equals("down")) {
                if (answer[1] > -1 * board[1]/2){
                    answer[1] -= 1;
                }
        }
    }
    return answer;
}
}